### PR TITLE
Feature extracted USAVars Dataset

### PR DESCRIPTION
**Pull request by nilsleh**
This PR adds a feature extracted version of USAVars. In the MOSAIKs paper, they show that RCF extracted features with RidgeRegression can outperform fully trained CNNs. In their code base they also have this feature extraced dataset version available. In order to not have to extract the same fixed RCF features in each epoch, this PR makes a feature extraced dataset version available to reduce the computational cost of running that experiment setting.

### CODE REVIEW

File: docs/api/datamodules.rst [hunk 1]
```diff
@@ -151,6 +151,11 @@ USAVars
 
 .. autoclass:: USAVarsDataModule
```
Comment (adamjstewart) [inline docs/api/datamodules.rst:152]: Let's just put it under this section, it's basically the same dataset

File: docs/api/datasets.rst [hunk 2]
```diff
@@ -351,6 +351,11 @@ USAVars
 
 .. autoclass:: USAVars
```
Comment (adamjstewart) [inline docs/api/datasets.rst:352]: Same

File: docs/api/non_geo_datasets.csv [hunk 3]
```diff
@@ -36,6 +36,7 @@ Dataset,Task,Source,# Samples,# Classes,Size (px),Resolution (m),Bands
 `Tropical Cyclone`_,R,GOES 8--16,"108,110",-,256x256,4K--8K,MSI
 `UC Merced`_,C,USGS National Map,"2,100",21,256x256,0.3,RGB
 `USAVars`_,R,NAIP Aerial,100K,-,-,4,"RGB, NIR"
+`USAVarsFeaturesExtracted`_,R,NAIP Aerial,100K,-,-,4,"RGB, NIR"
```
Comment (adamjstewart) [inline docs/api/non_geo_datasets.csv:39]: Same here, no need for a new row, the values are the same and the link will link to the above sections

**Review by adamjstewart — COMMENTED** [review]
Don't know much about the details but can dig a little deeper after I dig myself out of this paperwork hole.

**Comment by calebrob6** [comment]
Hey @nilsleh -- this is just running normal USAVars through an RCF and ResNet and exporting the results as a dataset?

**Comment by calebrob6** [comment]
I'm a bit hesitant as:
1. there are many parameters that one might want to play with to go from NAIP images in USAVars to the embedded versions (normalization, size of RCF kernels, number of RCF kernels, etc.) and this is simply choosing one such combination
2. caching the results of an RCF or ResNet18 embedding step isn't too difficult, so I don't think this saves much work

**Comment by adamjstewart** [comment]
This smells like #443

**Comment by nilsleh** [comment]
I don't have a strong opinion about it. I agree the embedding step is not difficult, but I thought I would just add the dataloading and module for it to be useable with torchgeo.

**Pull request closed without merging**